src/blockdigest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "blockdigest.h"
+#include "streams.h"
+#include "version.h"
+#include "uint256.h"
+#include "primitives/transaction.h"
+#include "utiltime.h"
+#include "utilstrencodings.h"
+
+namespace DigestFilter {
+
+std::vector<uint8_t> base_filter::outpoint_data(const COutPoint& outpoint) const
+{
+    CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
+    stream << outpoint;
+    return std::vector<uint8_t>(stream.begin(), stream.end());
+}
+
+std::vector<const std::vector<uint8_t>> base_filter::tx_data_vec(const CTransaction& tx) const
+{
+    std::vector<const std::vector<uint8_t>> tx_data;
+    const uint256& hash = tx.GetHash();
+    printf("- adding %s\n", hash.ToString().c_str());
+    tx_data.emplace_back(hash.begin(), hash.end());
+
+    for (unsigned int i = 0; i < tx.vout.size(); i++) {
+        printf("- adding OP %s:%d\n", hash.ToString().c_str(), i);
+        tx_data.emplace_back(outpoint_data(COutPoint(hash, i)));
```

Comment:
Well, so in theory this means you can iterate COutpoint(hash, 0...i) until you get a negative response then you know you've covered the txn (vout.size() < i), but I don't see what the benefit is of knowing that.